README.md

```diff
@@ -46,3 +46,28 @@ you can enable with the following import:
 ```scala
 import scala3encoders.given
 ```
+
+## Udf
+
+`Udf` or "user defined functions" enable spark to use own functions that process rows.
+The main routine for creating and registering those functions are 
+`spark.sql.functions.udf` and `SparkSession.register`. These cannot be replaced as simply as the encoders.
+The `udf` call itself is a generic function using `TypeTags` which are not available in Scala 3. The `register` is needed to make the calls available inside a `spark.sql` statement with a string identifier.
+
+To use spark-scala3 `Udf`:
+```scala
+import scala3udf.Udf
+```
+
+Instead of Scala 2’s `udf(lambda)`, you can use the call `Udf(lambda)`. Instead of calling `spark.register(myFun1)`, you can call either:
```

Comment:
hu - I thought it would possibly collide with the existing udf when `spark.sql.functions._` is imported - but it seems to work!
I will adapt the documentation + test

Reply:
the trick is to `import scala3udf.{Udf => udf}`